### PR TITLE
Stats & Reader: followers should direct to site-stream

### PR DESCRIPTION
**Pull request by samouri**
I recently noticed two places with references to Reader or Reader-like things that don't actually link to Reader.

The followers stats module located at: `/stats/insights/`
screen shot 2017-10-18 at 5 25 41 pm

And also the followers list located at: `/people/followers/`
screen shot 2017-10-18 at 5 27 38 pm

Stats names chosen:
1. `calypso_reader_stats_insights_followers_link_click`
2. `calypso_reader_people_followers_link_click`

Thanks @aaronyan for the help in naming!

todo:
- [x] to reader from insights
- [x] insights tracks
- [x] to reader from people
- [x] people tracks

### CODE REVIEW

**Comment by samouri** [comment]
To Test
-------

- [ ] verify ux of both /people and the stats insights page
- [ ] verify both tracks events fire

**Comment by fraying** [comment]
This is great, @samouri.

Bonus points (or maybe a separate PR) for adding follow links to the Followers page, maybe like so: 

follow-followers

(Pinging @jancavan for design review.)

It's always seemed weird to me that the list of followers on the Insights page has follow links, but the same info on the Followers page does not.

**Comment by jancavan** [comment]
Nice work @samouri! 

> Bonus points (or maybe a separate PR) for adding follow links to the Followers page, maybe like so:

Good call. I wanted to see how it'd look if we left-aligned the Follow links like in Reader:
screenshot 2017-10-19 11 17 22

But it looks weird. I think because the "Remove" link is always present (as opposed to Settings which only shows up after Following):

screenshot 2017-10-19 11 24 13

**Comment by jancavan** [comment]
I can confirm that the Author names now link to Reader and the events are being fired.

@samouri Did you happen to change this on the Stats Comment module as well? Or are Comments and Followers using the same component? It got updated in Comments too, which is good.

**Comment by samouri** [comment]
> Or are Comments and Followers using the same component? It got updated in Comments too, which is good.

Turns out the same component is reused :O.  I had no idea!

  I'll rename the tracks event

**Comment by samouri** [comment]
renamed the track event to `calypso_reader_stats_module_site_stream_link_click`

**Comment by jancavan** [comment]
@samouri @aaronyan Should two different events be fired so we can determine which click came from which module? Or is there a way to segment that?

**Comment by samouri** [comment]
@jancavan: decided to make it the same track but with a differentiating `moduleName` prop

File: client/my-sites/people/people-profile/index.jsx [hunk 1]
```diff
@@ -80,6 +81,8 @@ const PeopleProfile = React.createClass( {
 		return 'role-' + role;
 	},
 
+	handleLinkToReaderSiteStream: () => recordTrack( 'calypso_reader_people_followers_link_click' ),
```
Comment (blowery) [inline client/my-sites/people/people-profile/index.jsx]: hrm. this isn't really a reader event. probably `calypso_sites_people_followers_link_click` instead?

File: client/my-sites/stats/stats-list/stats-list-item.jsx [hunk 2]
```diff
@@ -216,8 +230,24 @@ const StatsListItem = React.createClass( {
...
+							recordTrack( 'calypso_reader_stats_module_site_stream_link_click', {
+								siteId,
+								module_name: this.props.moduleName,
+							} );
+						};
+					}
+				}
 				itemLabel = (
-					<a onClick={ this.preventDefaultOnClick } href={ data.link }>
+					<a onClick={ onClickHandler } href={ href }>
```
Comment (blowery) [inline client/my-sites/stats/stats-list/stats-list-item.jsx]: isn't `href` out of scope here?

File: client/my-sites/stats/stats-list/stats-list-item.jsx [hunk 3]
```diff
@@ -216,8 +230,24 @@ const StatsListItem = React.createClass( {
 			}
 
 			if ( data.link ) {
+				let href = data.link;
+				let onClickHandler = this.preventDefaultOnClick;
+				if ( this.isFollowersModule ) {
+					const siteId = this.getSiteIdForFollow();
+					if ( siteId ) {
+						href = '';
```
Comment (blowery) [inline client/my-sites/stats/stats-list/stats-list-item.jsx]: why do we erase the href?

Comment (samouri) [reply client/my-sites/stats/stats-list/stats-list-item.jsx]: i don't think we need to since i've added the `preventDefault()`

Comment (blowery) [reply client/my-sites/stats/stats-list/stats-list-item.jsx]: over in reader, we only prevent when it's a default (left) mouse click with no modifiers held. That way, a user can option-click or right click on the link and open the original URL in a new tab or window.

Comment (samouri) [reply client/my-sites/stats/stats-list/stats-list-item.jsx]: ah, ok ill update this

File: client/my-sites/people/people-profile/index.jsx (hunk 1 above)
Comment (samouri) [reply client/my-sites/people/people-profile/index.jsx]: sure

**Comment by samouri** [comment]
couple changes:
1. added a new prop to the tracks events that indicates if it was opened in a new tab or not
2. fixed a problem where we were giving html tags more properties than they accept
3. fixed an issue where i was nesting anchor tags

**Comment by blowery** [comment]
testing

**Comment by blowery** [comment]
tested out fine, LGTM

File: client/my-sites/people/people-profile/index.jsx [hunk 4]
```diff
@@ -77,6 +79,23 @@ class PeopleProfile extends React.PureComponent {
...
+
+		recordTrack( 'calypso_sites_people_followers_link_click', {
+			modifier_pressed: modifierPressed,
+		} );
+
+		if ( modifierPressed ) {
+			return;
+		}
+
+		const blogId = get( this.props.user, 'follow_data.params.blog_id', false );
```
Comment (blowery) [inline client/my-sites/people/people-profile/index.jsx]: should probably bail if we get `false` here

Comment (samouri) [reply client/my-sites/people/people-profile/index.jsx]: i already have the bail in calling code, but makes sense to have this fully independent

Comment (samouri) [reply client/my-sites/people/people-profile/index.jsx]: bail added! ready for final 👀

**Pull request merged**